core/src/main/java/org/apache/calcite/avatica/util/Sources.java

```diff
@@ -136,8 +136,7 @@ private static URL fileToUrl(File file) {
           // We need to encode path. For instance, " " should become "%20"
```

Comment:
nit:
Is this comment still needed ?

Reply:
No, probably not. Removed... but closer inspection here forced me to fix something else. This branch of the code is making a very "odd" URI/URL. It's hierarchical URI, with a relative and a scheme. I don't know if they're illegal under the RFC... but it's certainly weird. I had to subtly tweak things here to construct one by treating it as a non-heirarchical URL (args here are: scheme, scheme-specific-part, fragment). The behavior now matches the old code.